client/src/components/MyPage/LeftNav.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { Fragment, useMemo } from 'react';
+import guguStyled from '@/core/styled';
+
+import { NavLink } from '@/Router';
+import { Like, Buy, Review, QnA, Cart } from '@/static/assets/svg';
+
+import { greyBg1, greyButton, normalRadius, primary1 } from '@/static/style/common';
+
+const LeftNav = () => {
+  const navMenus = useMemo(() => {
+    return [
+      {
+        title: '쇼핑정보',
+        items: [
+          { path: '/cart', icon: <CustomCart />, title: '장바구니' },
+          { path: '/mypage/like', icon: <CustomLike />, title: '찜 목록' },
+          { path: '/mypage/order', icon: <CustomBuy />, title: '주문 목록' },
+        ],
+      },
+      {
+        title: '회원정보',
+        items: [
+          { path: '/mypage/QnA', icon: <CustomQnA />, title: '나의 상품문의' },
+          { path: '/mypage/review', icon: <CustomReview />, title: '나의 구매후기' },
+        ],
+      },
+    ];
+  }, []);
```

Comment:
고정된 값이면 컴포넌트 밖에 빼도 되지않아요? useMemo도 안써도될거같고 리렌더링 시 재할당 받을 필요도 없을거같아서 아래와 같이 해도 될거같습니다 :) 
```suggestion
  
const CustomCart = guguStyled(Cart)`
  ${getIconStyle()}
`;

const CustomLike = guguStyled(Like)`
  ${getIconStyle()}
`;

const CustomBuy = guguStyled(Buy)`
  ${getIconStyle()}
`;

const CustomReview = guguStyled(Review)`
  ${getIconStyle()}
`;

const CustomQnA = guguStyled(QnA)`
  ${getIconStyle()}
`;

const navMenus = [
  {
    title: '쇼핑정보',
    items: [
      { path: '/cart', icon: <CustomCart />, title: '장바구니' },
      { path: '/mypage/like', icon: <CustomLike />, title: '찜 목록' },
      { path: '/mypage/order', icon: <CustomBuy />, title: '주문 목록' },
    ],
  },
  {
    title: '회원정보',
    items: [
      { path: '/mypage/QnA', icon: <CustomQnA />, title: '나의 상품문의' },
      { path: '/mypage/review', icon: <CustomReview />, title: '나의 구매후기' },
    ],
  },
];
```

Reply:
icon에 CustomReview는 재정의 된 컴포넌트입니다.
밖에서 선언하게 되면 호이스팅 때문인지 에러가 나서요.. 아래와 같이 선언하고 useMemo를 걸어주었습니다!!

Reply:
엇 위에 처럼 하니깐 되던데요?!

Reply:
아! styled component를 위로 올리셨네요?

Reply:
아 저걸 위로올리기보단 저 배열을 아래로 내렸습니다 가독성이 좀 그러면 컴포넌트안에 const로 놔도 될거같기도합니다 👍

Reply:
근데 갑자기 든 생각인데 컴포넌트안에 const로 놓을빠엔 useMemo를 쓰는게 낫겠네유 👍